### PR TITLE
fix: derive a project key from non-Latin names

### PR DESCRIPTION
**Pull request by luantaraschi**
## Description

`generateProjectSlug` feeds the project key field as you type the name in the create-project modal, and the create button stays disabled while that key is empty. Two inputs come out wrong.

**A name written in a non-Latin script gets no key at all.** The regex is `[^A-Z0-9\s]`, so every letter outside ASCII is removed before the initials are taken:

```js
generateProjectSlug("тест")          // ""
generateProjectSlug("Проект Альфа")  // ""
generateProjectSlug("测试项目")        // ""
```

Empty key, disabled button, nothing on screen saying why. Seven of the seventeen locales in `i18n/` are non-Latin (`ru-RU`, `uk-UA`, `mk-MK`, `el-GR`, `hi-IN`, `ko-KR`, `zh-CN`), so this is reachable by the people those locales exist for.

**A leading separator eats an initial.** `split(/\s+/)` leaves an empty first entry, which still takes one of the three slots:

```js
generateProjectSlug(" Kaneo")             // "K"    (expected "KAN")
generateProjectSlug(" Alpha Beta Gamma")  // "AB"   (expected "ABG")
```

### Why this shape of fix

The API already answers this question. `toSlug` in `apps/api/src/column/controllers/create-column.ts` normalizes with NFKC and matches on `\p{L}\p{M}\p{N}` with the `u` flag, and `tests/api/column/to-slug.test.ts` has a case named "keeps non-Latin scripts instead of producing an empty slug". The web-side key generator was the one place still on `A-Z`. This lines the two up instead of inventing a third rule.

Words holding no letter or number are dropped, which covers the empty leading entry and also a word left holding only combining marks after punctuation is stripped.

### After

```js
generateProjectSlug("тест")               // "ТЕС"
generateProjectSlug("Проект Альфа")       // "ПА"
generateProjectSlug("测试项目")             // "测试项"
generateProjectSlug(" Kaneo")             // "KAN"
generateProjectSlug(" Alpha Beta Gamma")  // "ABG"
generateProjectSlug("ＡＢＣ")              // "ABC"  (NFKC, same as toSlug)
```

Unchanged: `"Kaneo"` is still `"KAN"`, `"Alpha Beta Gamma"` is still `"ABG"`, `"Sprint 2"` is still `"S2"`, `"[Alpha] Beta Gamma"` is still `"ABG"`, and a name with no letters or numbers still returns `""` so the button stays disabled exactly as before.

## Related Issue(s)

None open that I could find.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

## How Has This Been Tested?

- [x] Unit tests

Added `apps/web/src/lib/generate-project-id.test.ts`, which the module did not have. Seven cases: three fail on `main` and pass with the change, four pass on both sides so a harness that stopped exercising the function would not read as a fix.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective

## Additional Notes

`biome check` is clean on both files.

One caveat worth stating plainly: `pnpm install` does not complete on my Windows machine, because the workspace links need symlink privileges this box does not grant. I ran these tests with vitest pointed at the module rather than through `pnpm test`. The module imports nothing, so the run exercises the real code, but I have not run the rest of the web suite and CI should be the judge of that.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved project slug generation for international characters, full-width characters, combining marks, numbers, and punctuation.
  * More reliably removes invalid or empty content and returns an empty slug when no valid name content remains.
  * Improved task short-ID recognition across international, mixed alphanumeric, underscore, and hyphenated project keys.
  * Improved project search matching so wildcard characters are treated literally rather than matching unintended projects.

* **Tests**
  * Added coverage for slug generation, task short-ID matching, and safe search matching across supported character sets and edge cases.

### CODE REVIEW

**Comment by luantaraschi** [comment]
Both findings hold and I took both, in `0c0e634b`.

**Unicode-unsafe slicing.** Correct, and it was introduced by this PR: before, characters outside the basic multilingual plane were stripped along with everything non-ASCII, so nothing could be split. Now they survive, and `word[0]` on a surrogate pair returns half a character. The generator iterates by code point now:

```js
generateProjectSlug("????????")  // "??????"  (was one and a half characters)
generateProjectSlug("??lpha ??eta") // "????"
```

Also added the related case CodeRabbit raised: a word whose first code point is a combining mark now yields the first letter or number instead of the mark.

**ASCII-only short-id search.** Also correct, and it is the more useful of the two, since it makes the feature half-work rather than look wrong. `global-search.ts` matched `/^([A-Za-z][\w-]*)-(\d+)$/`, so a key like `??-23` never reached the slug-plus-number lookup and fell through to plain text search. The pattern now lives in `apps/api/src/search/task-short-id.ts` with the same Unicode rules the generator and `toSlug` use, and it still requires a leading letter and a trailing number, so `23-45` and `DEP-` stay unmatched.

Moving it out of the controller was to make it testable without the database and drizzle imports, which is what `tests/api/search/task-short-id-pattern.test.ts` covers now. That directory had no tests before.

12 tests across the two files, all green. `biome check` is clean.

File: apps/api/src/search/controllers/global-search.ts [hunk 1]
```diff
@@ -147,7 +148,7 @@ async function globalSearch(params: SearchParams): Promise<{
     : inArray(projectTable.workspaceId, accessibleWorkspaceIds);
 
   // Check if query matches short-id pattern (e.g. "DEP-23")
-  const shortIdMatch = query.match(/^([A-Za-z][\w-]*)-(\d+)$/);
+  const shortIdMatch = query.match(TASK_SHORT_ID_PATTERN);
```
[the remark this answers is not in the document]
Comment (luantaraschi) [reply apps/api/src/search/controllers/global-search.ts]: Confirmed against a real database rather than from memory. I ran Postgres 16 in a container with the keys `DEP`, `DEX`, `DE_`, `A_B` and `ACB`:

```
slug ILIKE 'A_B'     -> ACB, A_B      (2 rows)
slug ILIKE 'DE_'     -> DEP, DEX, DE_ (3 rows)
slug ILIKE 'A\_B'    -> A_B           (1 row)
```

So `A_B-23` could open a task in `ACB`, with `limit(1)` deciding which one, and it behaves the same through a bound parameter, which is how drizzle sends it. Fixed in `de090fd` by escaping `\`, `%` and `_` before the comparison. Escaping rather than switching to `lower() = lower()` keeps the case folding exactly where it was, in the database, so the only thing that changes is that wildcards stop being wildcards.

It predates this PR: `main` matched with `[\w-]*`, which already accepted `_`, and widening the class to `[\p{L}\p{N}\p{M}_-]` adds no new metacharacter, since neither `%` nor `\` is in it.

Your other point about normalization is in the same commit. `generateProjectSlug` normalizes to NFKC before a key is stored, so the query is normalized before it is matched, or a decomposed `ПА-23` never reaches the composed form on disk.

The escaping lives in `apps/api/src/search/like-pattern.ts` now, with tests in `tests/api/search/like-pattern.test.ts`, so it is covered without needing a database in CI.

File: apps/web/src/lib/generate-project-id.ts [hunk 2]
```diff
@@ -1,17 +1,42 @@
...
+  if (words.length === 0) {
+    return "";
+  }
+
+  // Iterate by code point, not by UTF-16 unit. A letter outside the basic
+  // multilingual plane is two units, so `word[0]` would hand back half a
+  // surrogate pair and `slice(0, 3)` would cut one in half.
   if (words.length === 1) {
-    return words[0].slice(0, 3);
+    return Array.from(words[0]).slice(0, 3).join("");
```
[the remark this answers is not in the document]
Comment (luantaraschi) [reply apps/web/src/lib/generate-project-id.ts]: Right, and it was inconsistent with the function's own other branch: `firstLetterOrNumber` already skips leading marks for a multi-word name, while the single-word path did not. Fixed in `de090fd`.

`generateProjectSlug("́alpha")` returned `́AL` before and returns `ALP` now. The concrete cost is the one your comment names: `TASK_SHORT_ID_PATTERN` wants `\p{L}` first and a mark is `\p{M}`, so the key the app generated could not be typed back to reach the task, which is the round trip this PR is about.

The test is in `generate-project-id.test.ts`, and it fails on the previous code with `expected '́AL' to be 'ALP'`.

**Comment by randoneering** [comment]
Thank you for the work! Devs should be taking a look at this soon.
